pageserver/src/tenant/timeline/import_pgdata/flow.rs

```diff
@@ -344,6 +319,68 @@ impl Flow {
     }
 }
 
+impl Plan {
+    async fn execute(
+        self,
+        timeline: Arc<Timeline>,
+        import_config: &TimelineImportConfig,
+        ctx: &RequestContext,
+    ) -> anyhow::Result<()> {
+        let mut work = FuturesOrdered::new();
```

Comment:
Why FuturesUnordered and not JoinSet?

Reply:
It's `FuturesOrdered`:
* not `FuturesUnordered` because we want completion to be ordered by the job index.
* not `JoinSet` because there's no ordered join set

We want completion to be ordered because it's the building block for check-pointing progress.
You can peek at https://github.com/neondatabase/neon/pull/11862 (WIP) to see where we are going.